build.sbt

```diff
@@ -44,7 +44,7 @@ def commonProject(id: String, projectPath: String, scalacVersion: String): Proje
     classLoaderLayeringStrategy := ClassLoaderLayeringStrategy.Flat,
 
     scalacOptions ++= Seq(
-      "-deprecation",
+      "-deprecation:false",
```

Comment:
Why are you disabling deprecation warnings?

Reply:
This is actually not a good setting when you are attempting to maintain binary compatibility in a library. It is good practice to use deprecation warnings to help library users migrate off of old code before it is removed. It is also good practice to have all code be unit tested (even deprecated code). So if you do both, then you are going to run into compiler errors with this flag turned on. I don't think it is a good setting for libraries. It is a good setting for the projects that use the libraries, but not the libraries themselves.